addon/models/node.js

```diff
@@ -105,5 +70,25 @@ export default OsfModel.extend({
     root: DS.belongsTo('node', {
         inverse: null
     }),
-    logs: DS.hasMany('logs')
+    logs: DS.hasMany('logs'),
+
+    save() {
+        // Some duplicate logic from osf-model#save needed to support
+        // contributor edits being saved through the node
+        var promise = this._super(...arguments);
+        var contributors = this.hasMany('contributors').hasManyRelationship;
+        if (contributors.hasData || contributors.hasLoaded) {
+            this.set(
+                '_dirtyRelationships.contributors.update',
+                contributors.members.list.filter(m => !m.record.get('isNew') && Object.keys(m.record.changedAttributes()).length > 0)
+            );
+            // Contributors are a 'real' delete, not just a de-reference
+            this.set(
+                '_dirtyRelationships.contributors.delete',
+                this.get('_dirtyRelationships.contributors.remove')
+            );
+            this.set('_dirtyRelationships.contributors.remove', []);
+        }
+        return promise;
```

Comment:
> it seems we call this.set for everything, regardless of whether the promise succeeds

True. But the _dirtyRelationships collection is (designed to be 😁 ) one-to-one, so _assuming the canonicalStates are updated correctly_, this is a non-issue.